src/types/checkbox.ts

```diff
@@ -0,0 +1,72 @@
+import type { LabelAttributes } from './input'
+
+export interface CheckboxProps {
+  /**
+   * Use v-model to bind the checked state of the underlying <input>.
+   * The v-model binds to the modelValue prop of the component and sets
+   * current checked state of the input.
+   * @required
+   */
+  modelValue: boolean
+
+  /**
+   * Will place label text to the right of the checkbox. Can also be slotted.
+   * @default ''
+   */
+  label?: string
+
+  /**
+   * KCheckbox has an instance of KLabel for supporting tooltip text.
+   * Use the labelAttributes prop to configure the KLabel's props.
+   * @default {}
+   */
+  labelAttributes?: LabelAttributes
+
+  /**
+   * Will place description text under the checkbox label. Can also be slotted.
+   * @default ''
+   */
+  description?: string
+
+  /**
+   * Use this prop to apply error styling to the component.
+   * @default false
+   */
+  error?: boolean
+}
+
+export interface CheckboxEmits {
+  /**
+   * Fired on change, returns the checked status of the checkbox (boolean).
+   */
+  change: [boolean]
```

Comment:
Let's name the parameters:

```suggestion
  change: [checked: boolean]
```